### ISSUE TITLE
Generic class with `ParamSpec` is inferring empty arguments as `Unknown`

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2692]
**Describe the bug**
Given the following code:

```python
from __future__ import annotations

from typing import Any, Callable, Concatenate, Generic, ParamSpec

P = ParamSpec('P')


class Context:
    ...


CommandHandler = Callable[Concatenate[Context, P], Any]


class Command(Generic[P]):
    def __init__(
        self,
        handler: CommandHandler[P],
        /,
        *,
        name: str | None = None,
        description: str | None = None,
    ) -> None:
        self.handler: CommandHandler[P] = handler
        self.name: str | None = name
        self.description: str | None = description


def handler_no_args(ctx: Context) -> None:
    ...


def handler_one_arg(ctx: Context, a: int) -> None:
    ...


cmd_no_args = Command(handler_no_args)
cmd_one_arg = Command(handler_one_arg)
```

`cmd_no_args` is inferred as `Command[Unknown]` while `cmd_one_arg` is inferred as `Command[(a: int)]`

**Expected behavior**
`cmd_no_args` should be inferred as `Command[()]`

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.194

**Comment by bryanforbes** [comment]
Another (probably related) issue I found (the following comes after the above code snippet):

```python
def create_handler() -> CommandHandler[[]]:
    def handler(ctx: Context) -> None:
        ...

    return handler
```

The return type has an error reported: `Declared return type, "(Context)", is partially unknown`

**Comment by erictraut** [comment]
Thanks for the bug report. Yes, these two cases are due to the same underlying bug. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.195, which I just published. It will also be included in the next release of pylance.
